pkg/workflows/internal/v2/sdkimpl/runtime.go

```diff
@@ -169,6 +172,33 @@ func (d *Runtime) RunInNodeMode(fn func(nodeRuntime sdk.NodeRuntime) *pb.SimpleC
 	})
 }
 
+func (d *Runtime) GenerateReport(
+	encodedPayload []byte,
+	encoderName, signingAlgo, hashingAlgo string,
+) sdk.Promise[*pb.ConsensusOutputs] {
+	d.nextCallId++ // Question: do I increment here?
+
+	observation := &pb.SimpleConsensusInputs{
+		Observation: &pb.SimpleConsensusInputs_Value{
+			Value: &valuespb.Value{
+				Value: &valuespb.Value_BytesValue{
+					BytesValue: encodedPayload,
+				},
+			},
+		},
+		Descriptors: &pb.ConsensusDescriptor{
+			EncoderName: encoderName,
+			SigningAlgo: signingAlgo,
+			HashingAlgo: hashingAlgo,
+		},
+	}
+	c := &consensus.Consensus{}
+
+	return sdk.Then(c.Simple(d, observation), func(result *pb.ConsensusOutputs) (*pb.ConsensusOutputs, error) {
+		return result, errors.New("not implemented")
```

Comment:
should I transform result into the output format expected for the write target?

Reply:
yes